tools/mute-logging/src/main/java/io/spine/testing/logging/MutingLoggerTap.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.testing.logging;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+import java.util.logging.Formatter;
+import java.util.logging.Handler;
+import java.util.logging.Logger;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Redirects the output of the associated JDK Logger to a memory stream.
+ */
+final class MutingLoggerTap {
+
+    /** The loggerName of the associated logger. */
+    private final String loggerName;
+
+    // Previous state of the Logger that we amend.
+    private boolean usedParentHandlers;
+    private @Nullable MemoizingStream memoizingStream;
+    private @Nullable Handler handler;
+    private @Nullable ImmutableList<Handler> previousHandlers;
+
+    /**
+     * Creates the muting tap for the log with the passed name.
+     */
+    MutingLoggerTap(String loggerName) {
```

Comment:
Is it the FQN name? If so — we can use `@FullyQualifiedName` annotation here, if not — let's document what kind of name we actually expect.

Reply:
Maybe it is better to pass `class` instance instead?

Reply:
We pass the name to allow some flexibility for possible future references to a package.